### PR TITLE
LF: Cleanup Fetches

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
- For the sake of explicitness, we rename Fetch Speedy command and Fetch update to FetchTemplate
- We replace some miuse of fetch_by_interface to fetch_interface
- Drop dead fetch_by_interface

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala [hunk 1]
```diff
@@ -625,38 +625,28 @@ private[lf] final class Compiler(
     //   let <tmplArg> = $fetch(tmplId) <coid>
     //       _ = $insertFetch(tmplId, false) coid [tmpl.signatories] [tmpl.observers] [tmpl.key]
     //   in <tmplArg>
-    topLevelFunction2(t.FetchDefRef(tmplId)) { (cidPos, tokenPos, env) =>
-      compileFetchBody(env, tmplId, tmpl)(cidPos, None, tokenPos)
+    topLevelFunction2(t.FetchTemplateDefRef(tmplId)) { (cidPos, tokenPos, env) =>
+      compileFetchTemplateBody(env, tmplId, tmpl)(cidPos, None, tokenPos)
     }
 
-  private[this] def compileFetchInterfaceBody(
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:632]: as it is used only once, I inline it.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala [hunk 2]
```diff
@@ -882,18 +874,6 @@ private[lf] final class Compiler(
       )
     }
 
-  private[this] def compileFetchByInterface(
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:885]: dead code.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala [hunk 3]
```diff
@@ -920,10 +900,10 @@ private[lf] final class Compiler(
       )
     case Command.ExerciseByKey(templateId, contractKey, choiceId, argument) =>
       t.ChoiceByKeyDefRef(templateId, choiceId)(s.SEValue(contractKey), s.SEValue(argument))
-    case Command.Fetch(templateId, coid) =>
-      t.FetchDefRef(templateId)(s.SEValue(coid))
-    case Command.FetchByInterface(interfaceId, templateId, coid) =>
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:925]: `FetchByInterface` is never used.

**Review by nickchapman-da — APPROVED** [review]
LGTM

File: daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala [hunk 4]
```diff
@@ -986,7 +986,7 @@ private[validation] object Typing {
         typeOfExerciseInterface(tpl, choice, cid, arg, guard)
       case UpdateExerciseByKey(tpl, choice, key, arg) =>
         typeOfExerciseByKey(tpl, choice, key, arg)
-      case UpdateFetch(tpl, cid) =>
+      case UpdateFetchTemplate(tpl, cid) =>
         typeOfFetch(tpl, cid)
```
Comment (akrmn) [inline daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala:990]: while we're at it, maybe we should rename `s/typeOfFetch/typeOfFetchTemplate/g` too

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala (hunk 3 above)
Comment (akrmn) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:911]: :fire:

**Review by akrmn — APPROVED** [review]
nice cleanup, thank you!

**Pull request merged**